### PR TITLE
[Solved #82] implement socket schema validation

### PR DESCRIPTION
**Pull request by supunTE**
Solved #82 

- Refactor socket protocols into an object
- Fix uuid package (which was not installed in package.json earlier)
- Implement auth token validation
- Make compound schemas for protocols (Join lobby, Create Lobby, Race Start)
- Inferred types from compound schemas
- Implement socket schema validation, created a map for protocols-schemas

### CODE REVIEW

File: libs/models/src/lib/sockets/protocols.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  SendTypeLog = 'SEND_TYPE_LOG',
+  UpdateTypeLogs = 'UPDATE_TYPE_LOGS',
+  Timeout = 'TIMEOUT',
+  ForceEnd = 'FORCE_END',
+  SendLeaderboard = 'SEND_LEADERBOARD',
+  ClearPlayer = 'CLEAR_PLAYER',
+  ResetLobby = 'RESET_LOBBY',
+}
+
+export const socketProtocols = {
```
Comment (dalanad) [inline libs/models/src/lib/sockets/protocols.ts]: Capitalize First letter

File: apps/server/src/utils/authtoken.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { v4 as uuidv4, validate, version } from 'uuid';
+
+export function generateAuthToken(): string {
```
Comment (dalanad) [inline apps/server/src/utils/authtoken.ts:8]: Put in comments

**Review by dalanad — CHANGES_REQUESTED** [review]

File: libs/models/src/lib/sockets/protocols.ts (hunk 1 above)
Comment (supunTE) [reply libs/models/src/lib/sockets/protocols.ts]: Updated 17c1260820a76a66b5e53d8da25f78656637797f

File: apps/server/src/utils/authtoken.ts (hunk 2 above)
Comment (supunTE) [reply apps/server/src/utils/authtoken.ts:8]: comments added
ee3669d7963e4244913ef2f9d707be0895d67cb1

**Review by dalanad — APPROVED** [review]

**Pull request merged**